### ISSUE TITLE
Fastlane uses faraday < 2.0 and forces to use slack-ruby-client < 2.0

### ISSUE THREAD

**Issue by havebeenfitz** [fastlane/fastlane#21334]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Fastlane is bound to the 1.* version of [faraday](https://github.com/lostisland/faraday) and the commonly used [slack-ruby-client](https://github.com/slack-ruby/slack-ruby-client) uses faraday 2.x+. It was ok to use the 1.x version of the latter, but now the Slack team deprecated some old API endpoints for sending messages and it seems like we can't use 1.x of slack-ruby-client anymore. Upgrading slack-ruby-client to 2.x version causes conflict in `Gemfile.lock` since it uses the newer major version of faraday

I'm not sure how painful it would be for fastlane to migrate to faraday 2.x, but it would be great to have it. Or maybe there are other options to have both bundled?

### How to reproduce

1. Add to Gemfile:

```
#ruby=3.1.0

source 'https://rubygems.org'

gem 'fastlane', '~> 2.213.0'
gem 'slack-ruby-client', '~> 2.1.0'
```

2. Run `bundle install`

----

**Expected result:** 

- Dependencies are installed sucessfuly 

**Actual result:**

- Dependencies failed to resolve

```
Bundler could not find compatible versions for gem "faraday":
  In Gemfile:
    fastlane (~> 2.213.0) was resolved to 2.213.0, which depends on
      faraday (~> 1.0)

    slack-ruby-client (~> 2.1.0) was resolved to 2.1.0, which depends on
      faraday (>= 2.0)
```

### Environment

fastlane 2.213.0 or any version

**Comment by gopito** [comment]
I think we have a PR https://github.com/fastlane/fastlane/pull/22270
But it needed to add some final steps in it 🙃

**Comment by okmanideep** [comment]
🚨 Bumping this up, since we might not have a way to upload build artifacts or such to slack from fastlane from [March 2025](https://api.slack.com/changelog/2024-04-a-better-way-to-upload-files-is-here-to-stay) after Slack discontinues `files_upload` API.

**Comment by JrGoodle** [comment]
@fastlaneTeam is there any way to get this prioritized? slack uploads will break on March 11

**Comment by okmanideep** [comment]
@JrGoodle https://github.com/fastlane/fastlane/pull/22270 seems like a big change. Chances are, it will go beyond the deadline. 

We have decided to create a upload to slack internal API for ourselves in one of our Node services and use the official Node sdk of Slack. The other alternative is to make three sequential API calls in the fastlane script as described in official documentation

> 1. Call `files.getUploadURLExternal`. The response will contain a URL that you can POST the contents of your file to.
> 2. POST the contents of your file to the URL returned in step 1. This can be done by sending the raw bytes, or can be a multipart form ended request. If all went well, Slack will respond with an HTTP 200.
> 3. Call `files.completeUploadExternal`. You can optionally specify a sharing destination for your file or files using this API's arguments. This finalizes the file upload started in step 1.

\- [Uploading files | Slack API Documentation](https://api.slack.com/messaging/files#uploading_files)

Hope this helps

**Comment by celciusjj** [comment]
any update about it?

**Comment by joshavant** [comment]
Landed here because I, too, am getting hamstrung while trying to use other gems that want to use a newer Faraday version and fastlane is holding me back.

**Comment by kblcuk** [comment]
I guess due to this (old faraday version) any fastlane command prints out warnings like the following:

```
/tmp
11:01:41 ❯ fastlane --version
fastlane installation at path:
/opt/homebrew/Cellar/fastlane/2.231.1/libexec/gems/fastlane-2.231.1/bin/fastlane
-----------------------------
[⠋] 🚀 Top level ::CompositeIO is deprecated, require 'multipart/post' and use `Multipart::Post::CompositeReadIO` instead!
Top level ::Parts is deprecated, require 'multipart/post' and use `Multipart::Post::Parts` instead!
[✔] 🚀
fastlane 2.231.1
```

**Comment by monchote** [comment]
@iBotPeaches Any chance this can be looked into? 🙏 There's a new security vulnerability affecting Faraday 2.14.0 and earlier. Patched in 2.14.1 https://www.cve.org/CVERecord?id=CVE-2026-25765

**Comment by iBotPeaches** [comment]
We are kinda blocked on any upgrades till we drop Ruby 2 support. It took a lot of work to drop 2.6, but once I drop 2.7 and 3.0. I think this is doable.
